src/Microsoft.ML.Core/Utilities/Hashing.cs

```diff
@@ -113,6 +114,7 @@ public static uint MurmurRound(uint hash, uint chunk)
         /// * 0x0800 to 0xFFFF : 1110xxxx 10xxxxxx 10xxxxxx
         /// NOTE: This MUST match the StringBuilder version below.
         /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Are you sure that the AggressiveInlining is needed / profitable on all these methods? E.g. this method is very large - I doubt that it is beneficial to aggresive inline it. 

The original issue was just about the Rotate.

Reply:
I tried removing AggressiveInlining attribute from murmur hash and the time for the benchmarks rises by 1.2 seconds for both netcoreapp2.1 and netcoreapp3.0

In the email thread @kouvel mentioned to add the attributes  to both the functions. Although the code that I added for Murmur hash in the related issue was little bit smaller version of the actual code

Reply:
Sorry I meant to say MurmurRound not MurmurHash. The last MurmurHash overload may be small enough to be worth inlining but the others look too large.

Reply:
Right, `AggressiveInlining` on method with loops is very questionable.